libs/application/templates/car-recycling/src/fields/VehiclesOverview/VehiclesOverview.tsx

```diff
@@ -16,92 +17,99 @@ import { formatText, getErrorViaPath } from '@island.is/application/core'
 import { Label } from '@island.is/application/ui-components'
 import { InputController } from '@island.is/shared/form-fields'
 import {
+  filterSelectedVehiclesFromList,
+  filterVehiclesList,
   getApplicationAnswers,
-  getApplicationExternalData,
 } from '../../lib/carRecyclingUtils'
 import { carRecyclingMessages, errorMessages } from '../../lib/messages'
 
 import { useFormContext } from 'react-hook-form'
 
-import { VehicleDto } from '../../shared/types'
+import { useDebounce } from 'react-use'
+import { useVehicles } from '../../hooks'
 import { FuelCodes } from '../../shared'
+import { VehicleDto } from '../../shared/types'
 
 const VehiclesOverview: FC<FieldBaseProps> = ({
   application,
   errors,
   error,
 }) => {
   const { formatMessage } = useLocale()
-
+  const { setValue } = useFormContext()
+  const [page, setPage] = useState(1)
+  const [permno, setPermno] = useState('')
   const [currentVehiclesList, setCurrentVehiclesList] = useState<VehicleDto[]>(
     [],
   )
   const [selectedVehiclesList, setSelectedVehiclesList] = useState<
     VehicleDto[]
   >([])
-  const [allVehiclesList, setAllVehiclesList] = useState<VehicleDto[]>([])
 
   const [canceledVehiclesList, setCanceledVehiclesList] = useState<
     VehicleDto[]
   >([])
 
-  const { setValue } = useFormContext()
-
-  const [page, setPage] = useState(1)
+  const {
+    getVehicles,
+    data: qlVehiclesData,
+    loading: loadingCurrent,
+  } = useVehicles()
+
+  const qlVehicleList = qlVehiclesData?.vehiclesListV2
+    .vehicleList as VehicleDto[]
+  const qlPaging = qlVehiclesData?.vehiclesListV2.paging
+
+  useDebounce(
+    () => {
+      if (permno !== '') {
+        getVehicles(1, permno)
+      } else {
+        // if on 1 page when starting to search, the page is cached so we need to call backend directly
+        // else on other pages we can just set page to 1
+        if (page === 1) {
+          getVehicles(1, '')
+        } else {
+          setPage(1)
+        }
+      }
+    },
+    500,
+    [permno],
+  )
 
-  const ITEMS_ON_PAGE = 8
-  const totalPages =
-    allVehiclesList.length > ITEMS_ON_PAGE
-      ? Math.ceil(allVehiclesList.length / ITEMS_ON_PAGE)
-      : 0
+  useEffect(() => {
+    getVehicles(page, '')
+  }, [page])
 
   useEffect(() => {
-    const { selectedVehicles, allVehicles, canceledVehicles } =
-      getApplicationAnswers(application.answers)
-    const { vehiclesList } = getApplicationExternalData(
-      application.externalData,
-    )
+    const vehicles = qlVehiclesData?.vehiclesListV2.vehicleList as VehicleDto[]
 
-    if (selectedVehicles.length > 0) {
-      setSelectedVehiclesList(selectedVehicles)
+    if (vehicles) {
+      setCurrentVehiclesList(
+        filterSelectedVehiclesFromList(selectedVehiclesList, vehicles),
```

Comment:
You can use `qlVehicleList` instead of creating and using `vehicles` constant

Reply:
Fixed